README.md

```diff
@@ -86,53 +86,79 @@ fastify.ready(err => {
   fastify.swagger()
 })
 ```
-
+<a name="api"></a>
 ## API
-#### register options
-```js
-{
-  swagger: {
-    info: {
-      title: String,
-      description: String,
-      version: String
-    },
-    host: String,
-    schemes: [ String ],
-    consumes: [ String ],
-    produces: [ String ],
-    securityDefinitions: Object
+<a name="register.options"></a>
+### register options
+<a name="modes"></a>
+#### modes
+`fastify-swagger` supports two registration modes `dynamic` and `static`:
+<a name="mode.dynamic"></a>
+* `dynamic` mode is the default one, if you use the plugin this way - swagger specification would be gathered from your routes definitions.
+  ```js
+  {
+    swagger: {
+      info: {
+        title: String,
+        description: String,
+        version: String
+      },
+      host: String,
+      schemes: [ String ],
+      consumes: [ String ],
+      produces: [ String ],
+      securityDefinitions: Object
+    }
   }
-}
-```
-*All the above parameters are optional.*  
-You can use all the properties of the [swagger specification](https://swagger.io/specification/), if you find anything missing, please open an issue or a pr!
+  ```
 
-<a name="options"></a>
-#### swagger options
-Calling `fastify.swagger` will return to you a JSON object representing your api, if you pass `{ yaml: true }` to `fastify.swagger`, it will return you a yaml string.
+  *All the above parameters are optional.*  
+  You can use all the properties of the [swagger specification](https://swagger.io/specification/), if you find anything missing, please open an issue or a pr!
+
+  Example of the `fastify-swagger` usage in the `dynamic` mode is available [here](examples/dynamic.js).
+<a name="mode.static"></a>
 
-If you pass `{ exposeRoute: true }` the plugin will expose the documentation with the following apis:
+* `static` mode should be configured explicitly. In this mode `fastify-swagger` serves given specification, you should craft it yourselfe.
+  ```js
+  {
+    mode: 'static',
+    specification: {
+      type: 'file',
```

Comment:
are there other types?

Reply:
right now not, but I thought that it will be possible to provide `url` or `file`

Reply:
then let's not have a  `type` at all.

Reply:
done